src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Memory.cs

```diff
@@ -0,0 +1,155 @@
+            memoryOfIntClass.Memory = new int[] { 1, 2, 3 };
+
+            string json = await Serializer.SerializeWrapper(memoryOfIntClass, new JsonSerializerOptions { WriteIndented = true });
+            Assert.Equal(@"{
+  ""Memory"": [
+    1,
+    2,
+    3
+  ]
+}".NormalizeLineEndings(), json);
+        }
+
+        [Fact]
+        public async Task SerializeReadOnlyMemoryOfTClassAsync()
+        {
+            ReadOnlyMemoryOfTClass<int> memoryOfIntClass = new ReadOnlyMemoryOfTClass<int>();
+            memoryOfIntClass.ReadOnlyMemory = new int[] { 1, 2, 3 };
+
+            string json = await Serializer.SerializeWrapper(memoryOfIntClass, new JsonSerializerOptions { WriteIndented = true });
+            Assert.Equal(@"{
+  ""ReadOnlyMemory"": [
+    1,
+    2,
+    3
+  ]
+}".NormalizeLineEndings(), json);
+        }
+
+        [Fact]
+        public async Task DeserializeMemoryOfTClassAsync()
+        {
+            string json = @"{
+  ""Memory"": [
+    1,
+    2,
+    3
+  ]
+}";
+            MemoryOfTClass<int> memoryOfIntClass = await Serializer.DeserializeWrapper<MemoryOfTClass<int>>(json);
+            AssertExtensions.SequenceEqual(new int[] { 1, 2, 3 }, memoryOfIntClass.Memory.Span);
+        }
+
+
+        [Fact]
+        public async Task DeserializeReadOnlyMemoryOfTClassAsync()
+        {
+            string json = @"{
+  ""ReadOnlyMemory"": [
+    1,
+    2,
+    3
+  ]
+}";
+            ReadOnlyMemoryOfTClass<int> memoryOfIntClass = await Serializer.DeserializeWrapper<ReadOnlyMemoryOfTClass<int>>(json);
+            AssertExtensions.SequenceEqual(new int[] { 1, 2, 3 }, memoryOfIntClass.ReadOnlyMemory.Span);
+        }
+
+        [Fact]
+        public async Task SerializeMemoryByteAsync()
+        {
+            Assert.Equal("\"VGhpcyBpcyBzb21lIHRlc3QgZGF0YSEhIQ==\"", await Serializer.SerializeWrapper<Memory<byte>>(s_testData.AsMemory()));
+            Assert.Equal("\"VGhpcyBpcyBzb21lIHRlc3QgZGF0YSEhIQ==\"", await Serializer.SerializeWrapper<ReadOnlyMemory<byte>>(s_testData.AsMemory()));
+        }
+
+        [Fact]
+        public async Task DeserializeMemoryByteAsync()
+        {
+            Memory<byte> memory = await Serializer.DeserializeWrapper<Memory<byte>>("\"VGhpcyBpcyBzb21lIHRlc3QgZGF0YSEhIQ==\"");
+            AssertExtensions.SequenceEqual(s_testData, memory.Span);
+
+            ReadOnlyMemory<byte> readOnlyMemory = await Serializer.DeserializeWrapper<ReadOnlyMemory<byte>>("\"VGhpcyBpcyBzb21lIHRlc3QgZGF0YSEhIQ==\"");
+            AssertExtensions.SequenceEqual(s_testData, readOnlyMemory.Span);
+        }
+
+        [Fact]
+        public async Task SerializeMemoryByteClassAsync()
+        {
+            MemoryOfTClass<byte> memoryOfByteClass = new MemoryOfTClass<byte>();
+            memoryOfByteClass.Memory = s_testData;
+
+            string json = await Serializer.SerializeWrapper(memoryOfByteClass, new JsonSerializerOptions { WriteIndented = true });
+            Assert.Equal(@"{
```

Comment:
You can use the `JsonTestHelpers.AssertJsonEqual` method to check if two pieces of JSON are equivalent even if they aren't byte-by-byte identical:

https://github.com/dotnet/runtime/blob/d59af2cf097acb100ad6a9cba652be34be6f4a2e/src/libraries/System.Text.Json/tests/Common/JsonTestHelper.cs#L43-L53

That is assuming there is reason to test for indented outputs.